### PR TITLE
Add normative text to proofs section.

### PR DESCRIPTION
**Pull request by msporny**
This PR rewords the proof section to use normative text for properties that are meant to be used across all data integrity proofs.

A future PR will move `proofValue` to a section around `DataIntegrityProof` and let cryptosuites decide how they want to represent their values. For example, some cryptosuites might use `jws` instead of `proofValue`... others might (though this would probably be a bad idea) break their proof values into multiple fields. I'm trying to get this section locked down before writing the one for `DataIntegrityProof`.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-integrity/pull/56.html" title="Last updated on Oct 3, 2022, 1:37 PM UTC (b6ad630)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-integrity/56/1a92bcf...b6ad630.html" title="Last updated on Oct 3, 2022, 1:37 PM UTC (b6ad630)">Diff</a>

### CODE REVIEW

File: index.html
Comment (msporny) [inline index.html]: Everything above here is deleted because it was duplicated content. All of this text still exists elsewhere in the spec.

File: index.html
Comment (msporny) [inline index.html]: Everything below here is deleted because it was duplicated content. All of this text still exists elsewhere in the spec.

File: index.html
Comment (dlongley) [inline index.html]:
```suggestion
The following attributes are defined for use in a <a>data integrity proof</a>:
```

File: index.html
Comment (dlongley) [inline index.html]: Removed `mandatory` because it is redundant with `required`.

```suggestion
Proof types determine what other fields are required to secure and
verify the proof.
```

File: index.html
Comment (dlongley) [inline index.html]:
```suggestion
link to a public key which includes cryptographic material that is used by a
```

File: index.html
Comment (dlongley) [inline index.html]: Added a full Web origin to examples so it's known as an acceptable value.

```suggestion
the creator of the proof. Examples of a domain parameter include:
`domain.example` (DNS domain), `https://domain.example:8443` (full Web origin),
`mycorp-intranet` (well-known text string), and
`b31d37d4-dd59-47d3-9dd8-c973da43b63a` (UUID).
```

File: index.html
Comment (dlongley) [inline index.html]: This definition is for a challenge. The `nonce` term is also defined, but is used to improve unlinkability when generating signal values (by randomizing them). It may be better to include the `nonce` within the `proofValue`, however, using whatever format best suites the cryptosuite -- so maybe we want a separate section for `nonce` that says that.

```suggestion
          <dt><dfn class="lint-ignore">challenge</dfn></dt>
```

File: index.html
Comment (dlongley) [inline index.html]:
```suggestion
value is used to mitigate replay attacks. Examples of a challenge value include:
```

**Review by dlongley — APPROVED** [review]

**Review by thereisnogabe — APPROVED** [review]

**Review by OR13 — APPROVED** [review]

File: index.html
Comment (TallTed) [inline index.html:424]: I wonder whether the URLs to which these example types map, or how that mapping is to be achieved in general, shouldn't also be included?

File: index.html
Comment (TallTed) [inline index.html]:
```suggestion
being misused by being applied to a purpose other than the one that was intended. For example, a
```

File: index.html
Comment (TallTed) [inline index.html]: It would be good to also include an example of misuse/misapplication and/or the harm potentially caused thereby.

File: index.html
Comment (TallTed) [inline index.html]:
```suggestion
The creator of a proof SHOULD include a string value that indicates its intended usage, which a verifier SHOULD use to ensure the proof was intended to be used by them. The
specification of the `domain` parameter is useful in challenge-response
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Just a few

File: index.html
Comment (msporny) [reply index.html:424]: Yes, it should be included. I have raised issue https://github.com/w3c/vc-data-integrity/issues/60 to make sure we have a discussion about this before I suggest some text. I believe the solution is straight forward, but we all know how that goes... better to get some preliminary buy-in via an issue before charging forth with some language that might be controversial.

File: index.html
Comment (msporny) [reply index.html]: Done in b6ad63027f86141740c936c7046f03a4458b28c8.

Comment (TallTed) [reply index.html]: Looks good at a quick read. We might try to take a page from existing X509 certs, which can be set to be usable for things like "use to sign email", "use to encrypt email", "use for login (e.g., with ssh)", etc. -- and the mis-step is clear in the error messages when you try to use an "email signer" cert for an ssh login.

**Review by TallTed — APPROVED** [review]

**Comment by msporny** [comment]
@decentralgabe please:

1. DO NOT merge PRs on specs for which you are not one of the Editors. In W3C WGs, only Chairs/Staff/Editors should be merging PRs on specs.
2. DO specify a disposition of comments, actions, and objections before merging (if you are the Editor).
3. DO NOT perform merge-commits, instead keep a linear commit history (either rebase-and-merge or squash-and-merge). Some will say merge-commits are fine, and those people usually have terrible Github commit hygiene :P. 
4. DO NOT delete other people's branches unless it is crystal clear that they are done with them (I was not done with this branch, and my local copy was hosed earlier today, so now I'm going to have to try and re-create this PR by cherry-picking a bunch of stuff off of main... which will take precious time and is prone to human error).

I'm going to go and try to fix main... potential Github refactoring spam incoming on this PR.

**Comment by thereisnogabe** [comment]
I'm sorry @msporny, my mistake. I [restored your branch](https://github.com/w3c/vc-data-integrity/tree/msporny-refactor-proofs). Happy to revert the commit if you would like.

for (3) this is a setting that can be changed per the github repo

**Comment by msporny** [comment]
Normative, multiple reviews, changes requested and made, no objections, merged by @decentralgabe (non-Editor) but confirmed by @msporny (Editor), commit history on `main` confirmed to match this PR.

**Comment by msporny** [comment]
> I [restored your branch](https://github.com/w3c/vc-data-integrity/tree/msporny-refactor-proofs). 

In the future, please don't do that. :) It has been known to confuse Echidna (W3C's spec publishing tool). Since we're pre-Echidna here, we should be fine... just a warning that doing this sort of thing on specs that are further along have side effects. I'll have to check to see if those side effects have kicked in here or not (I don't think they have).

> Happy to revert the commit if you would like.

Nope, please don't revert the commit! :)

It's fixed on main and reverting the commit will corrupt the PR. Just leave things as they are now, I'll take care of cleaning up/checking things. 

> for (3) this is a setting that can be changed per the github repo

Yes, I've turned off merge-commits for this repo.

**Comment by msporny** [comment]
Normative, multiple reviews, changes requested and made, no objections, merged by @decentralgabe (non-Editor) but confirmed by @msporny (Editor), commit history on `main` confirmed to match this PR.

**Pull request merged**
